bindings/python/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+pybind11_add_module(pybind11_BipedalLocomotion MODULE
+    # Main module
+    bipedal_locomotion.h
+    bipedal_locomotion.cpp
+    # Support files
+    Contacts.cpp
+    DCMPlanner.cpp
+    ParametersHandler.cpp
+    QuinticSpline.cpp
+    SwingFootPlaner.cpp
+    TimeVaryingDCMPlanner.cpp
+    )
+
+target_link_libraries(pybind11_BipedalLocomotion PUBLIC
+    BipedalLocomotion::ParametersHandler
+    BipedalLocomotion::Planners
+    BipedalLocomotion::System
+    )
+
+target_include_directories(pybind11_BipedalLocomotion PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+
+# The generated Python dynamic module must have the same name as the pybind11
+# module, i.e. `bindings`.
+set_target_properties(pybind11_BipedalLocomotion PROPERTIES
+    LIBRARY_OUTPUT_DIRECTORY "${BIPEDAL_LOCOMOTION_PYTHON_PACKAGE}"
+    OUTPUT_NAME "bindings")
+
+if(FRAMEWORK_COMPILE_tests)
+    add_subdirectory(tests)
+endif()
+
+# Output package is:
+#
+# bipedal_locomotion
+# |-- __init__.py (generated from main bindings CMake file)
+# `-- bindings.<cpython_extension>
+#
+install(TARGETS pybind11_BipedalLocomotion DESTINATION ${PYTHON_INSTDIR})
```

Comment:
Not part of the review, just a brainstorming not blocking the code to be merged in any way: I wonder if there is a way to install pybind11 bindings in a way that a library that depends on this one is able to write its own bindings that takes in input and return arguments defined in the pybind11 library of blf.

Reply:
This is a good point, maybe an advanced and rather uncommon use-case but I think that pybind11 to some extent would allow that. Let me think, in a05407e I changed the structure of the project and it could come handy for this use case. Maybe by making `bipedal_locomotion.h` public and installing it, I think that downstream projects could use our `bindings::Create*` functions to add blf types in their modules.

I would leave this out from this first implementation, but thanks for making me think about this possibility.